### PR TITLE
Autoencoder logged loss mean

### PR DESCRIPTION
**Pull request by diego-ponce**
Explain what this PR does.
* updates AutoencoderPipeline.fit to log the mean of the batches for each epoch
* added a trap for invalid num_epochs
* added a test for invalid epoch catching

### CODE REVIEW

File: numalogic/models/autoencoder/pipeline.py [hunk 1]
```diff
@@ -147,18 +149,18 @@ def fit(self, X: NDArray[float], y=None, log_freq: int = 5) -> "AutoencoderPipel
...
                 self.optimizer.zero_grad()
                 _, decoded = self._model(x_batch)
                 loss = self.criterion(decoded, x_batch)
                 loss.backward()
                 self.optimizer.step()
-
+                losses.append(loss.item())
             if epoch % log_freq == 0:
-                _LOGGER.info(f"epoch : {epoch}, loss_mean : {loss.item():.7f}")
+                _LOGGER.info(f"epoch : {epoch}, loss_mean : {np.array(losses).mean():.7f}")
```
Comment (ab93) [inline numalogic/models/autoencoder/pipeline.py:162]: Nitpick. 
We can just directly do `np.mean(losses)`

**Review by ab93 — COMMENTED** [review]
Minor comment. Thank you!

**Comment by ab93** [comment]
@diego-ponce could you fix the flake8 errors?

File: numalogic/models/autoencoder/pipeline.py (hunk 1 above)
Comment (diego-ponce) [reply numalogic/models/autoencoder/pipeline.py]: updated!

**Review by ab93 — APPROVED** [review]
Looks good!
